src/query/graphite/storage/m3_wrapper.go

```diff
@@ -126,12 +122,19 @@ func translateQuery(query string, opts FetchOptions) (*storage.FetchQuery, error
 
 func translateTimeseries(
 	ctx xctx.Context,
-	m3list m3ts.SeriesList,
+	result *storage.FetchResult,
 	start, end time.Time,
 ) ([]*ts.Series, error) {
+	m3list := result.SeriesList
 	series := make([]*ts.Series, len(m3list))
+	resolutions := result.Metadata.Resolutions
```

Comment:
Ah very interesting, yeah makes sense we have to return all the different resolutions per series for graphite queries hey.

Reply:
Yeah it's a bit annoying; was playing around with grouping resolutions together and sending a reso range or something but honestly don't think the slight improvement is worth the complexity